File: okpermission/src/main/java/com/wings/okpermission/java/PermissionFragment.java
package com.wings.okpermission.java;

import android.content.pm.PackageManager;
import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import java.util.ArrayList;
import java.util.List;


/**
 * @author AI
 */
public class PermissionFragment extends Fragment {

    private PermissionCallBack callBack;
    private static final int REQUEST_CODE = 1;

    public void requestPermission(PermissionCallBack callBack, String ... permissions){
        this.callBack = callBack;
        requestPermissions(permissions,REQUEST_CODE);
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        if (requestCode == REQUEST_CODE){
            List<String> deniedList = new ArrayList<>();
            for (int i = 0; i < grantResults.length; i++) {
                if (grantResults[i] != PackageManager.PERMISSION_GRANTED){
                    deniedList.add(permissions[i]);
                }
            }
            boolean allGranted = deniedList.isEmpty();
            callBack.onResult(allGranted,deniedList);
        }
    }
}

File: okpermission/src/main/java/com/wings/okpermission/kotlin/PermissionFragment.kt
package com.wings.okpermission.kotlin

import androidx.fragment.app.Fragment
import java.util.*

typealias PermissionCallBack = (Boolean, List<String>) -> Unit
private const val REQUEST_CODE : Int = 1

class PermissionFragment : Fragment() {

    private var callBack: PermissionCallBack? = null

    fun requestPermission(callBack: PermissionCallBack, vararg permissions: String) {
        this.callBack = callBack;
        requestPermissions(permissions, REQUEST_CODE)
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        if (requestCode == REQUEST_CODE){
            val deniedList = ArrayList<String>()
            for ((index,result) in grantResults.withIndex()){
                deniedList.add(permissions[index]);
            }
            val allGranted = deniedList.isEmpty();
            callBack?.let { it(allGranted,deniedList) }
        }
    }

}